npk/src/npk.rs

```diff
@@ -350,10 +359,64 @@ impl Builder {
     }
 }
 
+#[derive(Debug)]
+pub enum CompAlg {
+    Gzip,
+    Lzma,
+    Lzo,
+    Xz,
+}
+
+impl std::fmt::Display for CompAlg {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            CompAlg::Gzip => write!(f, "gzip"),
+            CompAlg::Lzma => write!(f, "lzma"),
+            CompAlg::Lzo => write!(f, "lzo"),
+            CompAlg::Xz => write!(f, "xz"),
+        }
+    }
+}
+
+impl FromStr for CompAlg {
+    type Err = Error;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "gzip" => Ok(CompAlg::Gzip),
+            "lzma" => Ok(CompAlg::Lzma),
+            "lzo" => Ok(CompAlg::Lzo),
+            "xz" => Ok(CompAlg::Xz),
+            _ => Err(Error::InvalidCompressionAlgorithm),
+        }
+    }
+}
+
+/// Squashfs Options
+pub struct SquashfsOpts {
+    /// The compression algorithm used (default gzip)
+    pub comp: CompAlg,
+    /// Size of the blocks of data compressed separately
+    pub block_size: u32,
+}
+
+impl Default for SquashfsOpts {
+    fn default() -> Self {
+        SquashfsOpts {
+            comp: CompAlg::Gzip,
+            block_size: 131072,
```

Comment:
Where does this number come from?

Reply:
That's the default value used by `mksquashfs`